### PR TITLE
Add progress message to tomographic reconstruction

### PR DESCRIPTION
**Pull request by yijiang1**

### CODE REVIEW

File: tomviz/python/Recon_ART.py [hunk 1]
```diff
@@ -23,6 +23,7 @@ def transform_scalars(self, dataset, Niter=1):
             raise RuntimeError("No scalars found!")
 
         # Generate measurement matrix
+        self.progress.message = 'Generating measurement matrix'
```
Comment (yijiang1) [inline tomviz/python/Recon_ART.py]: This message is not shown in the status bar. Is this a bug or something?

Comment (cjh1) [inline tomviz/python/Recon_ART.py]: Are you seeing the other message? I checked on linux and I am seeing them

Comment (yijiang1) [inline tomviz/python/Recon_ART.py]: Yeah I can see the message inside main loop. Maybe this is a Mac problem?

Comment (cjh1) [inline tomviz/python/Recon_ART.py]: So its just that particular message?

**Comment by Hovden** [comment]
Great. If it runs, merge it. I will test it out.

File: tomviz/python/Recon_ART.py [hunk 2]
```diff
@@ -47,6 +49,9 @@ def transform_scalars(self, dataset, Niter=1):
             f[:] = 0
             b = tiltSeries[s, :, :].transpose().flatten()
             for i in range(Niter):
+                self.progress.message = 'Slice No.' + \
+                    str(s + 1) + '/' + str(Nslice) + \
+                    ', iteration No.' + str(i + 1) + '/' + str(Niter)
```
Comment (cjh1) [inline tomviz/python/Recon_ART.py]: Its often simpler todo string substitution, so this expression can be replaced by

```python
 self.progress.message = 'Slice No.%d/%d, iteration No.%d/%d' % (s + 1, Nslice, i + 1, Niter)
```
Is nicely separates the format from the data.

File: tomviz/python/Recon_DFT.py [hunk 3]
```diff
@@ -51,6 +52,9 @@ def transform_scalars(self, dataset):
         for a in range(Nproj):
             if self.canceled:
                 return
+            self.progress.message = 'Tilt image No.' + \
+                str(a + 1) + '/' + str(Nproj)
+
```
Comment (cjh1) [inline tomviz/python/Recon_DFT.py]: This could be done with substitution as well.

File: tomviz/python/Recon_DFT.py [hunk 4]
```diff
@@ -29,6 +29,7 @@ def transform_scalars(self, dataset):
         pad_post = int(np.floor((Npad - Ny) / 2.0))
 
         # Initialization
+        self.progress.message = 'Initialization'
```
Comment (cjh1) [inline tomviz/python/Recon_DFT.py]: I will take a look, but this shouldn't prevent this PR from being merged.

**Comment by yijiang1** [comment]
Ok I've added messages to the other reconstruction methods.

**Pull request merged**